### PR TITLE
Physics boundaries

### PR DESCRIPTION
**Pull request by mrboisvert**
Fixed bugs where agents could walk through obstacles or would not be able to move at all after colliding with an obstacle.

Added test_boundaries_stop_movement unit test to test_game to test these bugs.

I also manually tested this by starting a game with the following player code:

class MyAgent(Agent):
    """Enter your code here ..."""
    angle = 20
    movement = 250
    hit_flag = False
    x = 20
    def run(self):
        if MyAgent.hit_flag:
            self.set_movement_direction(MyAgent.angle - 90)
            self.set_movement_speed(MyAgent.movement)
            MyAgent.x -= 1
            if MyAgent.x <= 0:
                MyAgent.hit_flag = False
                MyAgent.x = 20
            pass
        else:
            self.set_movement_speed(MyAgent.movement)
            self.set_movement_direction(MyAgent.angle)

    def on_obstacle_hit(self):
        MyAgent.angle += 90
        MyAgent.hit_flag = True

### CODE REVIEW

File: src/physics_engine.py [hunk 1]
```diff
@@ -71,7 +103,7 @@ def run_render_test(self):
         # use pygame for testing
         GRAY = (220, 220, 220)
         pygame.init()
-        size = 640, 240
+        size = 1074, 700
```
Comment (mayshukla) [inline src/physics_engine.py:106]: Could you use `PhysicsEngine.SPACE_WIDTH` and `PhysicsEngine.SPACE_HEIGHT` here?

**Review by mayshukla — CHANGES_REQUESTED** [review]

**Comment by mayshukla** [comment]
I made some minor suggestions. Otherwise, this looks good.

**Comment by mayshukla** [comment]
Also this agent code works great! We could use it in our example agent

File: src/agent.py [hunk 2]
```diff
@@ -162,6 +165,46 @@ def set_movement_speed(self, speed):
         angle = self.agent_state.velocity.get_angle()
         magnitude = speed if speed <= max_speed else max_speed
         self.agent_state.velocity = Vector2.from_angle_magnitude(angle, magnitude)
+        #print(self.agent_state.velocity)
+
+    def _clip_velocity(self):
+        """
+        Sets the agent's velocity to 0 if their current velocity would cause them to clip through an object they are colliding with.
+        """
+        for collision in self.collisions:
```
[the remark this answers is not in the document]
Comment (mrboisvert) [reply src/agent.py]: You're absolutely right that is definitely the more idiomatic way to do this. Made the suggested change.

**Pull request merged**